### ISSUE TITLE
Nova - one to many link on nested field 

### ISSUE THREAD

**Issue by vparpoil** [bluelibs/bluelibs#344]
Hi,
I want to define a collection with an array of members, storing the userId and their role :
```javascript
"members" : [
    {userId: "a", role: "admin"}, 
    {userId: "b", role: "user"}
]
```

I would like to define a one-to-many link using `members.userId`. It doesn't seem to work (an empty array comes back from queries).

The documentation suggests this should work for many to one relations (example given of `field: "profile.medicalProfileId"`)

I'm not sure it's a bug or a missing feature.
Thanks,

**Comment by theodorDiaconu** [comment]
Hello, unfortunately, we do not support this and it is intentional.
Previously we did support this, we called the $metaLinks. They are fine, but they do more harm.

Work relationally create another collection, this is what Nova does, it allows you to have a relational database and have fast queries over it even in NoSQL, also have a look at "deepSync" in CollectionBundle allows you to persist collections with links automatically. 

Field storage can indeed be nested by it is only an id or array of ids.
